wallet-new/src/Cardano/Wallet/Kernel/DB/AcidState.hs

```diff
@@ -122,21 +120,20 @@ newPending accountId tx = runUpdate' . zoom dbHdWallets $
 --
 -- * For every address encountered in the block outputs, create an HdAddress if it
 -- does not already exist.
---
--- TODO(@uroboros/ryan) Move BlockMeta inside PrefilteredBlock (as part of CBR-239: Support history tracking and queries)
-applyBlock :: (Map HdAccountId PrefilteredBlock, BlockMeta) -> Update DB ()
-applyBlock (blocksByAccount,meta) = runUpdateNoErrors $ zoom dbHdWallets $
+applyBlock :: Map HdAccountId PrefilteredBlock -> Update DB ()
+applyBlock blocksByAccount = runUpdateNoErrors $ zoom dbHdWallets $
     createPrefiltered
         initUtxoAndAddrs
         (\prefBlock -> zoom hdAccountCheckpoints $
-                           modify $ Spec.applyBlock (prefBlock,meta))
+                           modify $ Spec.applyBlock prefBlock)
         blocksByAccount
   where
     -- Accounts are discovered during wallet creation (if the account was given
     -- a balance in the genesis block) or otherwise, during ApplyBlock. For
     -- accounts discovered during ApplyBlock, we can assume that there was no
     -- genesis utxo, hence we use empty initial utxo for such new accounts.
-    initUtxoAndAddrs :: PrefilteredBlock -> (Utxo, [AddrWithId])
```

Comment:
This is unrelated to this PR per se, but it's something that I've been pondering for a while. Why is this assumption safe? I don't think it is, I think we need to prefilter the genesis UTxO here. After all, there is no guarantee that the account ID we discover doesn't have an initial balance in the genesis UTxO, is there?

Reply:
I think the new comment explains the empty initial Utxo better...

Reply:
Hmm, I think it can still be made clearer. Something like this:

NOTE: When we create the new wallet, we look at the genesis UTxO and create an initial balance for _all_ accounts that we recognize as ours. This means that when we later discover a new account that is also ours, it _cannot_ appear in the genesis UTxO, because if it did, we would already have seen it (the genesis UTxO is static, after all).

Something along those lines.

Reply:
that's nice, I'll use your version...